ClemBot.Bot/bot/custom_tag_prefix.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+import discord
+from discord.ext import commands
+
+import bot.bot_secrets as bot_secrets
+from bot.clem_bot import ClemBot
+from bot.errors import TagPrefixRequestError
+
+log = logging.getLogger(__name__)
+
+
+class CustomTagPrefix:
+
+    def __init__(self, *, default: str):
+        log.info(f'Setting default prefix too: "{default}""')
+        self.default = default
+
+    async def get_tag_prefix(self, bot: ClemBot, message: discord.Message):
+
+        tagprefixes = []
+
+        # Check if bot is in BotOnly mode, if it is we cant get custom tag prefixes
+        # so we have to fall back to self.default
+        if not bot_secrets.secrets.bot_only:
+            # noinspection PyBroadException
+            try:
+                # Try to grab the prefixes from the db, raise an error on failure
+                # and bailout, we cant respond to anything at the moment
+                tagprefixes = await bot.custom_tag_prefix_route.get_custom_tag_prefixes(message.guild.id, raise_on_error=True)
+            except Exception as e:
+                log.error('Custom tagprefix request failed with error: {error}', error=e)
+                raise TagPrefixRequestError('Requesting custom tagprefix from the api failed')
+
+        if len(tagprefixes) == 0:
+            tagprefixes = [self.default]
+
+        return commands.when_mentioned(bot, message) + tagprefixes
```

Comment:
Probably remove the call to `commands.when_mentioned(...)`, the bot isn't supposed to do tags when it's mentioned right? 
![image](https://user-images.githubusercontent.com/38477514/165831078-bb2b54fe-ac6b-4b0b-8fa0-ff49ca3ae751.png)

Reply:
I was mostly following most of the code structure of custom prefix and I had kept this as it is. 
So it should be only `return tagprefixes`?

Reply:
Yep!

Reply:
milo is correct here, we only use when_mentioned on commands